### PR TITLE
Implementation page updates

### PR DESCRIPTION
**Pull request by Menduist**
Each commit details it's changes, and some could be reverted if deemed inappropriate.

- Update Nim implementation details
There was some outdated / missing infos there

- Remove Zig from implementations 
Everything is missing

- Remove udp & utp transports 
No one actually implements thoses

- Remove libp2p-dcutr 
Seems to be the same thing as libp2p-hole-punching?

- Rename libp2p-swarm to libp2p-switch 
AFAIK swarm is just deprecated terminology for switch

- Change Stream Multiplexing order 
yamux (default one) -> mplex (common one) -> quic (rare one)

- Remove RendezVous C++ implementation 
AFAICT it doesn't exist?

The last big one is:
Sort implementations by number of implemented protocols

AFAICT, implementations are currently a bit random in their order.
The way I see it, we have two ways to sort them which would makes sense:
By TIOBE index of the language:
```
1 "Python"
2 "JVM"
4 "C++"
5 "C#"
7 "JavaScript"
12 "Go"
16 "Swift"
26 "Rust"
42 "Erlang"
70 "Nim"
```

By number of implemented modules:
```
     25 "Go"
     20 "Rust"
     20 "JavaScript"
     16 "Nim"
     13 "C++"
     12 "Swift"
      8 "JVM"
      7 "Python"
      5 "Erlang"
```
(generated by `jq 'inputs.libs[].langs[] | select(.status != "Missing").name' data/implementations/* | sort | uniq -c | sort -nr`)
I think the second one makes more sense, but I'm probably biased here :smile:

The sorting in the JSONs was done by hand, so errors might have slipped through, but I checked a few times and didn't see any

EDIT: preview by CI here: https://bafybeigafwk3yllahd3kmokivxpg5lvqfh4nt6vyxe23ydsyfyefsthloe.on.fleek.co/implementations/

### CODE REVIEW

**Review by p-shahi — APPROVED** [review]
> By number of implemented modules:

👍 I like this, it gives progressive visual indication of which implementations are more mature/maintained.

**Review by DannyS03 — COMMENTED** [review]
Many thanks @Menduist :) 

- Regarding Zig - We recently added placeholders (mainly to generate awareness that it exists) - but I don't believe the source code is open yet.
- Also in favor in ordering the implementations by number of implemented modules. Some things to consider:
  - We would have to update the order each time there are enough modules in an implementation that require an order change.
  - I'm thinking we *could* setup scripts that pull in modules implemented for each implementation and dynamically update the order (but not sure if this is feasible).

File: layouts/partials/implementation.html [hunk 1]
```diff
@@ -1,3 +1,5 @@
+{{ $impls := (slice "Go" "Rust" "JavaScript" "Nim" "C++" "Swift" "JVM" "Python" "Erlang") }}
```
Comment (Menduist) [inline layouts/partials/implementation.html:1]: I pushed a new system which allows easy re-ordering of languages here
(should have done this in the first place)

I'm not familiar enough with Hugo to generate this slice automatically, but at least it became very straightforward to update it semi-automatically with the `jq` command:
`jq 'inputs.libs[].langs | to_entries | .[] | select(.value.status != "Missing") | .key' data/implementations/* | sort | uniq -c | sort -nr | awk '{print $2}' | tr '\n' ' '`

Comment (DannyS03) [reply layouts/partials/implementation.html:1]: Nice:)

**Review by DannyS03 — APPROVED** [review]

**Pull request merged**